src/anndata/_core/index.py

```diff
@@ -130,7 +140,7 @@ def _fix_slice_bounds(s: slice, length: int) -> slice:
     return slice(start, stop, step)
 
 
-def unpack_index(index: Index) -> tuple[Index1D, Index1D]:
+def unpack_index(index: Index) -> tuple[IndexRest, IndexRest]:
```

Comment:
Shouldn’t this be `(Index without Ellipsis) -> tuple[Index1D, Index1D]`?

Reply:
No, because this can return `EllipsisType` on either of the entries

Reply:
my point is that it probably shouldn’t. see below

Reply:
In other words, `Index` can contain an ellipsis

Reply:
Ok, but @flying-sheep then if we want to use `unoack_index` everywhere, we need to let it accept tuples of length > 2